contracts/src/core/interfaces/IEigenDAAddressDirectory.sol

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+interface IEigenDAAddressDirectory {
+    event AddressSet(string name, address indexed value);
+
+    /// @notice Sets the address of a contract.
+    function setAddress(string memory name, address value) external;
+
+    /// @notice Gets the address by keccak256 hash of the name.
+    function getAddress(bytes32 key) external view returns (address);
```

Comment:
do we really need this? Who would use this? Should we add the key as a field in the emitted event if so?